### PR TITLE
Add support to *scan parameters to ShardedJedis and JedisCluster

### LINKED ISSUE #1225: HSCAN, SSCAN and ZSCAN not available on BinaryCluster
### Expected behavior

`HSCAN`, `SSCAN` and `ZSCAN` should be available on `BinaryCluster` 

### Actual behavior

`HSCAN`, `SSCAN` and `ZSCAN` are only available with `String` signatures which breaks the key (and also the slot hash calculation) if keys are binary (`byte[]`).

We noticed the difference in signatures while testing the jedis Cluster integration in [Spring Data Redis](https://github.com/spring-projects/spring-data-redis/blob/master/src/main/java/org/springframework/data/redis/connection/jedis/JedisClusterConnection.java#L1553).

#### Jedis version:

2.8.0

### PR DESCRIPTION
**Pull request by marcosnils**
Fixes #1225

### CODE REVIEW

File: src/main/java/redis/clients/jedis/BinaryJedisCluster.java [hunk 1]
```diff
@@ -1751,4 +1751,67 @@ public Double execute(Jedis connection) {
...
     }.runBinary(key);
   }
+
+  @Override
+  public ScanResult<Map.Entry<byte[], byte[]>> hscan(final byte[] key, final byte[] cursor) {
+    return new JedisClusterCommand<ScanResult<Map.Entry<byte[], byte[]>>>(connectionHandler,
+        maxRedirections) {
+      @Override
+      public ScanResult<Map.Entry<byte[], byte[]>> execute(Jedis connection) {
+        return connection.hscan(cursor, key);
```
Comment (HeartSaVioR) [inline src/main/java/redis/clients/jedis/BinaryJedisCluster.java]: @marcosnils two parameters should be switched.

**Comment by HeartSaVioR** [comment]
Left one comment, otherwise seems good to me.

**Comment by marcosnils** [comment]
@HeartSaVioR thanks for checking carefully, clearly copy/paste is not my strongest ability :tongue:

**Comment by marcosnils** [comment]
@HeartSaVioR fixes applied per your comment and also found some other copy/pasting errors. Can you please re-check?

**Comment by HeartSaVioR** [comment]
@marcosnils Great! LGTM.

**Comment by mp911de** [comment]
:+1:

**Comment by marcosnils** [comment]
I'll downmerge to 2.9 later as I'm on mobile

**Comment by marcosnils** [comment]
@mp911de this breaks backwards compatibility so it's not likely that will get into 2.8.x. In any case we will try to release 2.9 this week

**Comment by mp911de** [comment]
No hurry. Not sure what you mean by breaking compatibility as this PR does not change/remove any method signatures.

**Comment by marcosnils** [comment]
@mp911de it thought this breaks binary compatibility but it doesn't. We're good to downmerge to 2.8.x.

https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html

Thx for pointing out

**Comment by marcosnils** [comment]
Downmerged to 2.8 and 2.9 respectively

**Comment by mp911de** [comment]
Any update on the 2.8.1 release date?

**Comment by marcosnils** [comment]
@mp911de today :)

**Comment by mp911de** [comment]
:+1:

**Pull request merged**